nft-minter/src/nft_module.rs

```diff
@@ -0,0 +1,211 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use crate::{
+    common_storage::{BrandId, BrandInfo, Tag},
+    unique_id_mapper::{UniqueId, UniqueIdMapper},
+};
+
+const NFT_AMOUNT: u32 = 1;
+const NFT_ISSUE_COST: u64 = 50_000_000_000_000_000; // 0.05 EGLD
+const ROYALTIES_MAX: u32 = 10_000; // 100%
+const VEC_MAPPER_FIRST_ITEM_INDEX: usize = 1;
+
+const MAX_BRAND_ID_LEN: usize = 50;
+static INVALID_BRAND_ID_ERR_MSG: &[u8] = b"Invalid Brand ID";
+
+#[elrond_wasm::module]
+pub trait NftModule:
+    crate::common_storage::CommonStorageModule
+    + crate::admin_whitelist::AdminWhitelistModule
+    + crate::nft_attributes_builder::NftAttributesBuilderModule
+    + crate::royalties::RoyaltiesModule
+{
+    #[payable("EGLD")]
+    #[endpoint(issueTokenForBrand)]
+    fn issue_token_for_brand(
+        &self,
+        brand_id: BrandId<Self::Api>,
+        media_type: ManagedBuffer,
+        royalties: BigUint,
+        max_nfts: usize,
+        mint_start_epoch: u64,
```

Comment:
Not really rounds - it would be hard for a non-tech user to compute the nonce/round at a desired time/date. But your point is valid - having start epoch will also create a lot of volume if multiple collections will start minting in the same date. We can actually use a timestamp in UTC and compare it with the block timestamp. The FE could give a user a nice UI to select when he wants the minting to start

Reply:
Changed to timestamp... and fixed to actually check said timestamp on buyRandomNft